_includes/scripts.njk

```diff
@@ -40,3 +40,229 @@
+        descriptors[prop] = type === "bool"
+          ? reflectBool(name)
+          : reflectString(name);
+      }
+      Object.defineProperties(klass.prototype, descriptors);
+      klass.observedAttributes = Object.keys(schema);
+    }
+
+    function dashToCamel(attr) {
+      return attr.replace(/-([a-z])/g, (_, c) => c.toUpperCase());
+    }
+
+    function reflectString(attr) {
+      return {
+        get() {
+          return this.getAttribute(attr);
+        },
+        set(v) {
+          if (v == null) {
+            this.removeAttribute(attr);
+          } else {
+            this.setAttribute(attr, v);
+          }
+        }
+      };
+    }
+
+    function reflectBool(attr) {
+      return {
+        get() {
+          return this.hasAttribute(attr);
+        },
+        set(v) {
+          if (v) {
+            this.setAttribute(attr, "");
+          } else {
+            this.removeAttribute(attr);
+          }
+        }
+      };
+    }
+
+    class HumanDateElement extends HTMLElement {
+      static attrs = {
+        "abbreviate-months": "bool",
+        "always-months": "bool",
+        clickable: "bool",
+        datetime: "string",
+        raw: "bool",
+      };
+
+      constructor() {
+        super();
+        this.attachShadow({ mode: 'open' });
+        this._timer = null;
+      }
+
+      connectedCallback() {
+        this._render();
+      }
+
+      disconnectedCallback() {
+        this._clearTimer();
+      }
+
+      attributeChangedCallback(name, oldVal, newVal) {
+        if ((name === 'datetime' || name === 'raw' || name === 'abbreviate-months' || name === 'always-months') && oldVal !== newVal) {
+          this._render();
+        }
+        if (name === 'clickable') {
+          if (newVal !== null) {
+            this.addEventListener('click', this._toggleRaw);
+          } else {
+            this.removeEventListener('click', this._toggleRaw);
+          }
+        }
+      }
+
+      _toggleRaw = (ev) => {
+        if (ev.altKey) {
+          return; // alt/option + click to avoid toggle
+        }
+        const sel = typeof this.shadowRoot.getSelection === "function"
+          ? this.shadowRoot.getSelection()
+          : document.getSelection();
+        if (sel && sel.rangeCount && !sel.isCollapsed) {
+          return;
+        }
+        this.toggleAttribute("raw");
+      };
+
+      _clearTimer() {
+        if (this._timer) {
+          clearTimeout(this._timer);
+          this._timer = null;
+        }
+      }
+
+      _parseDate(raw) {
+        if (!raw || typeof raw !== 'string') {
+          return null;
+        }
+        const normalized = raw.replace(/\s+/, 'T');
+        const d = new Date(normalized);
+        if (!isNaN(d.getTime())) {
+          return d;
+        }
+        const d2 = new Date(raw);
+        return isNaN(d2.getTime()) ? null : d2;
+      }
+
+      _format(now, dt) {
+        const ms = now - dt;
```

Comment:
It looks like you're missing timezone handling. Dates in the data are UTC. It's now 17:48 in UTC, 19:48 locally. It should say "just now", but it says "2 hours ago". If I put my local current time in the `created_at` (which would be 2 hours from now) it does say "just now".

<img width="762" height="174" alt="Scherm­afbeelding 2025-08-21 om 19 47 48" src="https://github.com/user-attachments/assets/8f670f64-9a6c-4134-b617-f6a5d56c4997" />

I wonder, if we're now doing all date presentation stuff client-side anyway... can't we just present everything in local time? (And change the comment about that in the footer as well)

Reply:
Local iso-ish formats can be confusing:

<img width="197" height="66" alt="image" src="https://github.com/user-attachments/assets/2ef896fc-7178-4e4e-ba56-524c481f68bf" />

However, the humanized format is local.  It would show "today, 17:29".

Reply:
> now it is anything or nothing

As far as I can tell, if you’re running in GitHub actions it’s UTC. The build timestamp of the site definitely is. It must be some kind of timezone deterministically, otherwise we should not be mentioning times and even dates become questionable. Or I don’t understand your point.

Reply:
Oh, you’re saying they’re UTC “by convention” because we know they are, but since the timezone is stripped they’re not UTC explicitly? I must say I find that really inelegant about our spec… what use is a timestamp if you don’t know what timezone it’s in? Maybe we should make a v5 schema 😉